pandas/tests/reshape/test_reshape.py

```diff
@@ -608,6 +608,69 @@ def test_get_dummies_all_sparse(self):
         )
         tm.assert_frame_equal(result, expected)
 
+    @pytest.mark.parametrize(
+        "values",
+        [
+            ["baz", "zoo"],
+            np.array(["baz", "zoo"]),
+            pd.Series(["baz", "zoo"]),
+            pd.Index(["baz", "zoo"]),
+        ],
+    )
+    def test_get_dummies_with_list_like_values(self, values):
```

Comment:
Is this testing new behavior? Are we not already testing this?

Reply:
In the file `reshape.py` there was no check on dtype passed to one of the parameters `columns` and so on the issue it was mentioned add this check. Hence, after adding the test I added these tests specifically to `columns` parameter. And regarding doubt whether this test was already present, I checked all tests again and there was no test to check the inputs to parameter `columns` and so I referred to a similar test as mentioned in PR and added it. Please let me know if you spotted a similar test for `get_dummies` which I might have missed. I hope this clears it.

Reply:
It looks like https://github.com/pandas-dev/pandas/blob/367670e80ff008d77c791437a7a5df3e1abe012b/pandas/tests/reshape/test_reshape.py#L90 tests with both columns and dtype?

Reply:
The line you are have pointed to does not check for string inputs to the `columns` parameters also it does not test for other list-like objects passed to it.

Reply:
I agree this test seems superfluous; this PR should just check for invalid values in columns.

Reply:
I will remove this test. It would be great help if could you elaborate/point me to a similar test for my reference? Because as far i understand if the value is not list-like the error message will be raised because of this PR and if invalid value (i.e. value not in data columns) is passed  to columns then error will be raised because value will not be present.

Reply:
> if the value is not list-like the error message will be raised because of this PR and if invalid value (i.e. value not in data columns) is passed to columns then error will be raised because value will not be present.

This test is redundant with https://github.com/pandas-dev/pandas/blob/367670e80ff008d77c791437a7a5df3e1abe012b/pandas/tests/reshape/test_reshape.py#L90.

We want to keep your test `test_get_dummies_with_string_values`, which is testing the code you've added in this PR.